src/transfer-manager.ts

```diff
@@ -396,12 +396,14 @@ export class TransferManager {
         ...options.passthroughOptions,
       };
 
-      passThroughOptionsCopy.destination = filePath;
+      passThroughOptionsCopy.destination = filePath
+        .split(path.win32.sep)
+        .join(path.posix.sep);
       if (options.prefix) {
-        passThroughOptionsCopy.destination = path.join(
-          options.prefix,
-          passThroughOptionsCopy.destination
-        );
+        passThroughOptionsCopy.destination = path
+          .join(options.prefix, passThroughOptionsCopy.destination)
+          .split(path.win32.sep)
+          .join(path.posix.sep);
```

Comment:
Maybe the prefix here should be split first to preserve the `\` for POSIX (in case customers really want it)

```suggestion
          .posix.join(...options.prefix.split(path.sep), passThroughOptionsCopy.destination);
```